### PR TITLE
Breaking changes changelog, 2023q3 version

### PR DESCRIPTION
**Pull request by tzarc**
## Description

As per the title.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/ChangeLog/20230827.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+* Unicode: move keycode aliases to a separate header ([#21613](https://github.com/qmk/qmk_firmware/pull/21613))
+* Bluetooth: driver naming cleanups ([#21624](https://github.com/qmk/qmk_firmware/pull/21624))
+* Remove old OLED API code ([#21651](https://github.com/qmk/qmk_firmware/pull/21651))
+* haptic: further naming cleanups ([#21682](https://github.com/qmk/qmk_firmware/pull/21682))
+* Simplfy RGB/LED matrix effect logic ([#21703](https://github.com/qmk/qmk_firmware/pull/21703))
+* OLED: driver naming cleanups ([#21710](https://github.com/qmk/qmk_firmware/pull/21710))
+
+CLI:
+* Add *_MATRIX_LED_COUNT generation/validation ([#19515](https://github.com/qmk/qmk_firmware/pull/19515))
+* Revert "Add *_MATRIX_LED_COUNT generation/validation" ([#21109](https://github.com/qmk/qmk_firmware/pull/21109))
```
Comment (Nebuleon) [inline docs/ChangeLog/20230827.md:152]: This PR got reverted and reissued
```suggestion
```

File: docs/ChangeLog/20230827.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+* Consolidate `KEYBOARD_OUTPUT`+`KEYMAP_OUTPUT`=>`INTERMEDIATE_OUTPUT` ([#21272](https://github.com/qmk/qmk_firmware/pull/21272))
+* Chibios USB: Take into account if host wants remote wakeup or not ([#21287](https://github.com/qmk/qmk_firmware/pull/21287))
+* Fix anchor IDs for some API references ([#21345](https://github.com/qmk/qmk_firmware/pull/21345))
+* Pixel fractal: Set minimum middle column value ([#21365](https://github.com/qmk/qmk_firmware/pull/21365))
+* Fix ili9xxx inversion opcode entry ([#21422](https://github.com/qmk/qmk_firmware/pull/21422))
+* Relocate backlight drivers ([#21444](https://github.com/qmk/qmk_firmware/pull/21444))
+* Fixup STM32-DFU ([#21447](https://github.com/qmk/qmk_firmware/pull/21447))
+* keycode aliases: work around ChibiOS ch.h include guard ([#21497](https://github.com/qmk/qmk_firmware/pull/21497))
+* Fix compilation error when Split Watchdog enabled ([#21543](https://github.com/qmk/qmk_firmware/pull/21543))
+* Revert " Fix compilation error when Split Watchdog enabled" ([#21572](https://github.com/qmk/qmk_firmware/pull/21572))
```
Comment (Nebuleon) [inline docs/ChangeLog/20230827.md:262]: This PR got reverted
```suggestion
```

File: docs/ChangeLog/20230827.md (hunk 1 above)
Comment (tzarc) [reply docs/ChangeLog/20230827.md:162]: This is the full changelist, inclusive of any failed commits and reverts. `qmk generate-develop-pr-list -h`.

File: docs/ChangeLog/20230827.md (hunk 2 above)
Comment (tzarc) [reply docs/ChangeLog/20230827.md:272]: This is the full changelist, inclusive of any failed commits and reverts. `qmk generate-develop-pr-list -h`.

**Review by dunk2k — APPROVED** [review]

**Comment by tzarc** [comment]
> Several sections in the documentation still encourage users to submit personal keymaps. I've [modified what I've found accordingly](https://github.com/lesshonor/qmk_firmware/commit/cf5bce0a45), but am not sure if you would want to incorporate that into this PR or if I should submit it separately.

Ahhhh you beat me to it! Will incorporate your changeset.

**Comment by tzarc** [comment]
Just needs the list of moved keyboards -- jscotto's stuff threw a spanner in the works.

File: docs/contributing.md [hunk 3]
```diff
@@ -67,7 +67,7 @@ We have a few different types of changes in QMK, each requiring a different leve
 * Separate PRs into logical units. For example, do not submit one PR covering two separate features, instead submit a separate PR for each feature.
 * Check for unnecessary whitespace with `git diff --check` before committing.
 * Make sure your code change actually compiles.
-  * Keymaps: Make sure that `make keyboard:your_new_keymap` does not return any errors.
+  * Keymaps: Make sure that `make keyboard:_keymap` does not return any errors.
```
Comment (noroadsleft) [inline docs/contributing.md:70]: Tiny typo fix.

```suggestion
  * Keymaps: Make sure that `make keyboard:keymap` does not return any errors.
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: docs/ChangeLog/20230827.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+| hfdkb/keyboard_sw/k83                 | inland/kb83                         |
+| idb_60                                | idb/idb_60                          |
+| kamigakushi                           | jaykeeb/kamigakushi                 |
+| kbdfans/kbd67mkiirgb                  | kbdfans/kbd67/mkiirgb               |
+| modelh                                | ibm/model_m/modelh                  |
+| vinta                                 | coarse/vinta                        |
+
+### Remove encoder in-matrix workaround code ([#20389](https://github.com/qmk/qmk_firmware/pull/20389)) :id=remove-encoder-in-matrix-workaround-code
+
+Some keyboards "hacked" encoder support into spare slots in the key matrix in order to interoperate with VIA. This workaround is no longer necessary, and the code has been removed. If you have a keyboard that uses this workaround, you will need to update your keymap to use the new encoder map API instead.
```
Comment (fauxpark) [inline docs/ChangeLog/20230827.md:45]:
```suggestion
Some keyboards "hacked" encoder support into spare slots in the key matrix in order to interoperate with VIA. This workaround is no longer necessary, and the code has been removed. If you have a keyboard that uses this workaround, you will need to update your keymap to use the new [Encoder Map](feature_encoders.md#encoder-map) API instead.
```

File: docs/ChangeLog/20230827.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+### Driver naming consolidation ([#21551](https://github.com/qmk/qmk_firmware/pull/21551), [#21558](https://github.com/qmk/qmk_firmware/pull/21558), [#21580](https://github.com/qmk/qmk_firmware/pull/21580), [#21594](https://github.com/qmk/qmk_firmware/pull/21594), [#21624](https://github.com/qmk/qmk_firmware/pull/21624), [#21710](https://github.com/qmk/qmk_firmware/pull/21710)) :id=driver-naming-consolidation
+
+In most circumstances this won't affect users -- only keyboard designers with currently-unmerged boards. The only users affected are people who have modified existing keyboards in order to add/modify haptics, lighting, or bluetooth -- and only if the base keyboard did not configure them already. Driver naming has been modified to be lowercase.
+
+RGBLight (`RGBLIGHT_DRIVER` / `rgblight.driver`):
+
+| Old    | New    |
+|--------|--------|
+| WS2812 | ws2812 |
+| APA102 | apa102 |
```
Comment (fauxpark) [inline docs/ChangeLog/20230827.md:69]:
```suggestion
| `WS2812` | `ws2812` |
| `APA102` | `apa102` |
```

File: docs/ChangeLog/20230827.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+LED Matrix (`LED_MATRIX_DRIVER` / `led_matrix.driver`):
+
+| Old         | New         |
+|-------------|-------------|
+| IS31FL3731  | is31fl3731  |
+| IS31FL3742A | is31fl3742a |
+| IS31FL3743A | is31fl3743a |
+| IS31FL3745  | is31fl3745  |
+| IS31FL3746A | is31fl3746a |
+| CKLED2001   | ckled2001   |
```
Comment (fauxpark) [inline docs/ChangeLog/20230827.md:80]:
```suggestion
| `IS31FL3731`  | `is31fl3731`  |
| `IS31FL3742A` | `is31fl3742a` |
| `IS31FL3743A` | `is31fl3743a` |
| `IS31FL3745`  | `is31fl3745`  |
| `IS31FL3746A` | `is31fl3746a` |
| `CKLED2001`   | `ckled2001`   |
```

File: docs/ChangeLog/20230827.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+| IS31FL3733  | is31fl3733  |
+| IS31FL3736  | is31fl3736  |
+| IS31FL3737  | is31fl3737  |
+| IS31FL3741  | is31fl3741  |
+| IS31FL3742A | is31fl3742a |
+| IS31FL3743A | is31fl3743a |
+| IS31FL3745  | is31fl3745  |
+| IS31FL3746A | is31fl3746a |
+| CKLED2001   | ckled2001   |
+| WS2812      | ws2812      |
```
Comment (fauxpark) [inline docs/ChangeLog/20230827.md:97]:
```suggestion
| `AW20216`     | `aw20216`     |
| `IS31FL3731`  | `is31fl3731`  |
| `IS31FL3733`  | `is31fl3733`  |
| `IS31FL3736`  | `is31fl3736`  |
| `IS31FL3737`  | `is31fl3737`  |
| `IS31FL3741`  | `is31fl3741`  |
| `IS31FL3742A` | `is31fl3742a` |
| `IS31FL3743A` | `is31fl3743a` |
| `IS31FL3745`  | `is31fl3745`  |
| `IS31FL3746A` | `is31fl3746a` |
| `CKLED2001`   | `ckled2001`   |
| `WS2812`      | `ws2812`      |
```

File: docs/ChangeLog/20230827.md [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+
+OLED (`OLED_DRIVER`):
+
+| Old     | New     |
+|---------|---------|
+| SSD1306 | ssd1306 |
```
Comment (fauxpark) [inline docs/ChangeLog/20230827.md:103]:
```suggestion
| `SSD1306` | `ssd1306` |
```

File: docs/ChangeLog/20230827.md [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+
+Haptic (`HAPTIC_DRIVER`):
+
+| Old      | New      |
+|----------|----------|
+| DRV2605L | drv2605l |
```
Comment (fauxpark) [inline docs/ChangeLog/20230827.md:109]:
```suggestion
| `DRV2605L` | `drv2605l` |
| `SOLENOID` | `solenoid` |
```

File: docs/ChangeLog/20230827.md [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+
+Bluetooth (`BLUETOOTH_DRIVER` / `bluetooth.driver`):
+
+| Old         | New          |
+|-------------|--------------|
+| BluefruitLE | bluefruit_le |
+| RN42        | rn42         |
```
Comment (fauxpark) [inline docs/ChangeLog/20230827.md:116]:
```suggestion
| `BluefruitLE` | `bluefruit_le` |
| `RN42`        | `rn42`         |
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**